bitsandbytes/cuda_setup/env_vars.py

```diff
@@ -50,3 +51,17 @@ def get_potentially_lib_path_containing_env_vars() -> Dict[str, str]:
         for env_var, value in os.environ.items()
         if is_relevant_candidate_env_var(env_var, value)
     }
+
+def restore_original_system_env_vars(env_vars: Dict[str, str]):
+    """
+        Use the `sudo -i env` command to obtain the original environment variables of the system, including CONDA_PREFIX, LD_LIBRARY_PATH...
+        Then replace the environment variables obtained from the current shell, to obtain correct and complete environment variables
+    """
+    output = subprocess.check_output(['sudo', '-i', 'env'])
```

Comment:
This should only be executed if we are running on a linux system (bitsandbytes will soon support windows).

Reply:
I thought that `sudo -i env` does not require a password, but it does. I think with that we cannot have this in the main code-path that is executed by everyone. 

Is there a way to detect if the python program was started with sudo? I think if we could detect this and then apply this fix conditionally dependent on that this would be a good solution.

Otherwise, can you think of different solutions to this issue that does not require calling sudo?

Reply:
I use `os.geteuid()` to ensure that the python program is started with `sudo`. If the program was started with `sudo`, this value will be 0 (the user ID of the superuser).

Reply:
And I use `find_executable('sudo')` to ensure the system supports sudo command. When running on a Windows system, this code will not be executed.